spinalcordtoolbox/scripts/sct_compute_ascor.py

```diff
@@ -0,0 +1,157 @@
+    """
+    df_sc = pd.read_csv(csa_sc)
+    df_canal = pd.read_csv(csa_canal)
+    df_ascor = pd.DataFrame()
+    # Loop across rows in dataframe
+    for idx in range(len(df_sc)):
+        ascor_value = df_sc['MEAN(area)'].iloc[idx] / df_canal['MEAN(area)'].iloc[idx]
+        row = [df_sc['Filename'].iloc[idx],
+               df_canal['Filename'].iloc[idx],
+               df_sc['Slice (I->S)'].iloc[idx],
+               df_sc['VertLevel'].iloc[idx],
+               df_sc['DistancePMJ'].iloc[idx],
+               ascor_value]
+        df_ascor = pd.concat([df_ascor, pd.DataFrame([row], columns=INDEX_COLUMNS)], ignore_index=True)
+    printv(f"Computed aSCOR for {len(df_ascor)} rows.", 1, 'normal')
+    # Save to csv
+    save_ascor_to_csv(df_ascor, fname_out, append)
+
+
+def save_ascor_to_csv(df_ascor, fname_out, append=False):
+    """
+    Saves the aSCOR DataFrame to a CSV file.
+
+    :param pandas.DataFrame: df_ascor : The DataFrame containing aSCOR results to be saved.
+    :param str: fname_out : Path to the output CSV file.
+    :param bool: append : If True, appends `df_ascor` to the existing CSV file. If False, overwrites the file. Default is False.
+    :return None
+
+    """
+    if append and os.path.exists(fname_out):
+        dataframe_old = pd.read_csv(fname_out, index_col=INDEX_COLUMNS)
+        df_ascor = pd.concat([dataframe_old.reset_index(), df_ascor], ignore_index=True)
+    df_ascor.to_csv(fname_out, index=False)
+
+
+def main(argv: Sequence[str]):
+    parser = get_parser()
+    arguments = parser.parse_args(argv)
+    verbose = arguments.v
+    set_loglevel(verbose=verbose, caller_module_name=__name__)    # values [0, 1, 2] map to logging levels [WARNING, INFO, DEBUG]
+
+    # Load input and output filenames
+    fname_sc_segmentation = get_absolute_path(arguments.i_SC)
+    fname_canal_segmentation = get_absolute_path(arguments.i_canal)
+    img_sc = Image(fname_sc_segmentation).change_orientation('RPI')
+    img_canal = Image(fname_canal_segmentation).change_orientation('RPI')
+    if not img_sc.data.shape == img_canal.data.shape:
+        raise ValueError(f"Shape mismatch between spinal cord segmentation [{img_sc.data.shape}],"
+                         f" and canal segmentation [{img_canal.data.shape}]). "
+                         f"Please verify that your spinal cord and canal segmentations were done in the same space.")
+    fname_out = arguments.o
+    temp_folder = TempFolder(basename="process-segmentation")
+    path_tmp = temp_folder.get_path()
+    # Run sct_process_segmentation on spinal cord segmentation
+    printv("Running sct_process_segmentation on spinal cord segmentation...", verbose, 'normal')
+    # Remove '-i-SC', '-o', and '-i-canal' arguments from argv before passing to sct_process_segmentation
+    process_seg_argv = [a for i, a in enumerate(argv) if argv[i-1] not in ['-i-SC', '-o', '-i-canal'] and a not in ['-i-SC', '-o', '-i-canal']]
```

Comment:
If we can assume that every `argv` will be passed in pairs (`-arg value`), then we can use `zip` to process the pairs, and then we can use `itertools.chain.from_iterable` to re-combine the pairs of arguments:

```python
process_seg_argv = itertools.chain.from_iterable([
    [arg, value] for arg, value in zip(argv[::2], argv[1::2])
    if arg not in ['-i-SC', '-i-canal', '-o']
])
```

Even though this is no longer a one-liner, I think this might read a little more clearly?

Reply:
done here https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/5007/commits/fa66f70f4a99fe5c47e4a4d69bd03fd6504efa0c